### PR TITLE
Update unit tests for password strategies

### PR DESCRIPTION
**Pull request by jessieay**
* Move specs into a more logical directory
* Remove subject/before/let statements
* Extract common code into scoped methods

### CODE REVIEW

File: spec/password_strategies/bcrypt_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
+require "spec_helper"
+
+describe Clearance::PasswordStrategies::BCrypt do
+  describe "#password=" do
+    it "encrypts the password into encrypted_password" do
+      stub_bcrypt_password
+
+      model_instance.password = password
```
Comment (jessieay) [inline spec/password_strategies/bcrypt_spec.rb]: good one! Fixed in fa52b81816b81ef5ffb3b826e96effeac080190a

File: spec/password_strategies/blowfish_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+require "spec_helper"
+
+describe Clearance::PasswordStrategies::Blowfish do
+  describe "#password=" do
+    context "when the password is set" do
+      it "does not initialize the salt" do
+        model_instance.salt = salt
```
Comment (jessieay) [inline spec/password_strategies/blowfish_spec.rb]: Done 99b79b21e9342a1aab1246454686897ffbc8d7c0

File: spec/support/fake_model_without_password_strategy.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+end
+
+class UniquenessValidator < ActiveModel::Validator
+  def validate(record)
+  end
+end
+
+RSpec.configure do |config|
+  config.include FakeModelWithoutPasswordStrategy
```
Comment (jessieay) [inline spec/support/fake_model_without_password_strategy.rb]: good call - fixed in 4b3dd05095b366fe8c1e0fe5c1fd59e8d219c4a9

File: spec/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,209 @@
+require "spec_helper"
+
+describe User do
```
Comment (jessieay) [inline spec/user_spec.rb]: `spec/support/clearance.rb` - updated descriptions to make it clearer! :rainbow:

**Comment by derekprior** [comment]
looks good!

**Pull request merged**